manifests/setup.pp

```diff
@@ -5,18 +5,17 @@
 #
 class jail::setup () {
 
-  package { 'iocage':
-    ensure => installed,
+  package { $jailmanager:
```

Comment:
This variable is out of scope here.  Instead of creating the various jail::$jailmanager classes, you could probably just include the $jailmanager variable in the setup class and set a sane default in the hiera data.  I can be more specific if I'm not making sense.

Reply:
That would help; I've never used heira, and as such, I'm functioning from "Do everything in manifests and templates" land.

Reply:
I'm sorry for being slow.

If we had a `hiera.yaml` file in the root of this repo that looks a bit like this one https://github.com/xaque208/puppet-unbound/blob/master/hiera.yaml then we can set `jail::setup::jailmanager: 'iocage'` in `data/common.yaml`.  This implies also that we only need a `common` section in the `hiera.yaml` file.  Let me know if that helps.  If that can get cleaned up a bit then I think this is merge worthy.  Does that all make sense?

Reply:
No, *I'm* sorry for being slow.
This seemed to have gotten hidden under all my other notifications :(

It makes sense, as far as it can to someone who's never used heira before.

I don't want to try writing one right now and fucking it all up, but I can clean up this PR so it'll work with a properly written file, if you still want to merge this :)

Reply:
I can handle the hiera bits, so no worry.  I'll link you the change.

Reply:
Awesomepossum, thanks!
When the hiera bit is done, I'll make the appropriate change here if something needs updating :)

Reply:
The hiera bits are in now.  You should now be able to add a parameter to the `jail::setup` class that you can read from hiera.  In `data/common.yaml` you could set something like `jail::setup::manager: 'iocage'` which would choose icoage as a default, then users would be able to override this variable in their own heira data, so you could do `jail::setup::manager: 'iocell'` for to select the correct package.